lib/ASRouter.jsm

```diff
@@ -289,6 +320,38 @@ const MessageLoaderUtils = {
             provider.id,
             options.dispatchToAS
           );
+        } else if (["cfr", "cfr-fxa"].includes(provider.id)) {
+          // Both "cfr" and "cfr-fxa" require the Fluent file for l10n, currently
+          // they share the same attachment of the record "cfr-${version}-${locale}",
+          // e.g. for "en-US" with version "v1", the record ID is "cfr-v1-en-US".
+
+          const locale = Services.locale.appLocaleAsLangTag;
+          const recordId = `cfr-${RS_FLUENT_VERSION}-${locale}`;
+
+          // Skip the downloaded if it's already fetched.
+          if (MessageLoaderUtils._remoteAttachmentFetched.includes(recordId)) {
+            return messages;
+          }
+
+          const kinto = new KintoHttpClient(
+            Services.prefs.getStringPref(RS_SERVER_PREF)
+          );
+          const record = await kinto
+            .bucket(RS_MAIN_BUCKET)
+            .collection(RS_COLLECTION_L10N)
+            .getRecord(recordId);
+          if (record && record.data) {
```

Comment:
I would invert this test in order to add a warning or throw...
```js
if (!record || !record.data) {
   console.warn("No language package found");
}
```

Reply:
Good point.